docs/sphinx/api/builtin_ops.rst

```diff
@@ -0,0 +1,255 @@
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // H = (1 / sqrt(2)) * | 1   1 |
+    //                     | 1  -1 |
+    h(q);
+
+:code:`r1`
+---------------------
+
+This is an arbitrary rotation about the |1> state.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // R1(λ) = | 1     0    |
+    //         | 0  exp(iλ) |
+    r1(std::numbers::pi, q);
+
+:code:`rx`
+---------------------
+
+This is an arbitrary rotation about the X axis.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // Rx(θ) = |  cos(θ/2)  -isin(θ/2) |
+    //         | -isin(θ/2)  cos(θ/2)  |
+    rx(std::numbers::pi, q);
+
+:code:`ry`
+---------------------
+
+This is an arbitrary rotation about the Y axis.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // Ry(θ) = | cos(θ/2)  -sin(θ/2) |
+    //         | sin(θ/2)   cos(θ/2) |
+    ry(std::numbers::pi, q);
+
+:code:`rz`
+---------------------
+
+This is an arbitrary rotation about the Z axis.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // Rz(λ) = | exp(-iλ/2)      0     |
+    //         |     0       exp(iλ/2) |
+    rz(std::numbers::pi, q);
+
+:code:`s`
+---------------------
+
+This operation applies to its target a rotation by π/2 about the Z axis.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // S = | 1   0 |
+    //     | 0   i |
+    s(q);
+
+:code:`t`
+---------------------
+
+This operation applies to its target a π/4 rotation about the Z axis.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // T = | 1      0     |
+    //     | 0  exp(iπ/4) |
+    t(q);
+
+:code:`swap`
+---------------------
+
+This operation swaps the states of two qubits.
+
+.. code-block:: cpp
+
+    // Apply the unitary transformation
+    // Swap = | 1 0 0 0 |
+    //        | 0 0 1 0 |
+    //        | 0 1 0 0 |
+    //        | 0 0 0 1 |
+    swap(q1, q2);
+
+Measurements on Qubits
+=============================
+
+:code:`mz`
+---------------------
+
+This operation measures a qubit with respect to the computational basis, 
+i.e. it projects the state of that qubit onto the eigenvectors of the Pauli-Z matrix.
+This is a non-linear transformation, and no template overloads are available.
+
+.. code-block:: cpp
+
+    mz(q);
+
+:code:`mx`
+---------------------
+
+This operation measures a qubit with respect to the Pauli-X basis, 
+i.e. it projects the state of that qubit onto the eigenvectors of the Pauli-X matrix.
+This is a non-linear transformation, and no template overloads are available.
+
+.. code-block:: cpp
+
+    mx(q);
+
+:code:`my`
+---------------------
+
+This operation measures a qubit with respect to the Pauli-Y basis, 
+i.e. it projects the state of that qubit onto the eigenvectors of the Pauli-Y matrix.
+This is a non-linear transformation, and no template overloads are available.
+
+.. code-block:: cpp
+
+    my(q);
+
```

Comment:
It seems like these definitions are already present in the MLIR documentation which can be generated through tablegen. Can't that be used?

Reply:
See my comment in the PR description. Yes, I would prefer to do that, but couldn't find a good way to do that as is.